src/component/parser/json/converter/ProcessConverter.ts

```diff
@@ -187,12 +190,15 @@ export default class ProcessConverter {
       } else if (kind == ShapeBpmnElementKind.GROUP) {
         shapeBpmnElement = buildShapeBpmnGroup(this.convertedElements, this.parsingMessageCollector, bpmnElement as TGroup, parentId);
       } else {
-        // eslint-disable-next-line @typescript-eslint/ban-ts-comment -- We know that the 'text' & 'name' fields are not on all types, but it's already tested
-        // @ts-ignore
-        const name = kind === ShapeBpmnElementKind.TEXT_ANNOTATION ? bpmnElement.text : bpmnElement.name;
-        // eslint-disable-next-line @typescript-eslint/ban-ts-comment -- We know that the 'instantiate' field is not on all types, but it's already tested
-        // @ts-ignore
-        shapeBpmnElement = new ShapeBpmnElement(bpmnElement.id, name, kind, parentId, bpmnElement.instantiate);
+        let name: string;
```

Comment:
**nitpick:** The complexity is now too high (see my remark below as well)

![image](https://github.com/process-analytics/bpmn-visualization-js/assets/27200110/8a9540a6-2416-4f85-9db9-2f47ac16f2ca)